client/src/main/java/io/pravega/client/tables/KeyValueTable.java

```diff
@@ -98,197 +122,147 @@
+     *            (i.e., this retrieves the latest value for a key).
+     * @return A CompletableFuture that, when completed, will contain the requested result. If no such {@link TableKey}
+     * exists, this will be completed with a null value.
      */
-    CompletableFuture<TableEntry<KeyT, ValueT>> get(@Nullable String keyFamily, @NonNull KeyT key);
+    CompletableFuture<TableEntry> get(@NonNull TableKey key);
 
     /**
-     * Gets the latest values for a set of Keys that do belong to the same Key Family.
+     * Gets the latest values for a set of {@link TableKey}s.
      *
-     * @param keyFamily (Optional) The Key Family for all requested Keys.
-     * @param keys      An {@link Iterable} of Keys to get values for.
+     * @param keys An {@link Iterable} of @link TableKey} to get values for.
      * @return A CompletableFuture that, when completed, will contain a List of {@link TableEntry} instances for the
-     * requested keys. The size of the list will be the same as keys.size() and the results will be in the same order
-     * as the requested keys. Any keys which do not have a value will have a null entry at their index.
+     * requested keys. The size of the list will be the same as {@code keys.size()} and the results will be in the same
+     * order as the requested keys. Any keys which do not have a value will have a null entry at their index.
      */
-    CompletableFuture<List<TableEntry<KeyT, ValueT>>> getAll(@Nullable String keyFamily, @NonNull Iterable<KeyT> keys);
+    CompletableFuture<List<TableEntry>> getAll(@NonNull Iterable<TableKey> keys);
 
     /**
-     * Creates a new Iterator over all the {@link TableKey}s in this {@link KeyValueTable} that belong to a specific
-     * Key Family.
+     * Creates a new Iterator for {@link TableKey}s in this {@link KeyValueTable}. This is preferred to {@link #entryIterator}
+     * if all that is needed is the {@link TableKey}s as less I/O is involved both server-side and between the server
+     * and client.
      *
-     * @param keyFamily     The Key Family for which to iterate over keys.
-     * @param maxKeysAtOnce The maximum number of {@link TableKey}s to return with each call to
-     *                      {@link AsyncIterator#getNext()}.
-     * @param state         (Optional) An {@link IteratorState} that represents a continuation token that can be used to
-     *                      resume a previously interrupted iteration. This can be obtained by invoking
-     *                      {@link IteratorItem#getState()}. A null value will create an iterator that lists all keys.
-     * @return An {@link AsyncIterator} that can be used to iterate over all the Keys in this {@link KeyValueTable} that
-     * belong to a specific Key Family.
+     * @param maxKeysAtOnce The maximum number of {@link TableKey}s to return with each call to {@link AsyncIterator#getNext()}.
```

Comment:
I know this as pagination, and I'm wondering if we should call it that way.

Reply:
What would sound better as an arg? I thought about a better name, but adding the word "page" implies UI concerns to me. Really, I want the user to be in control of how many items to return at once, for whatever reasons the user may have.

Reply:
What about: `iterationSize`, `maxIterationSize`, `iterationPage`, `maxIterationPage`, `iterationLimit`.

Reply:
I like `maxIterationSize`. Fixed.